### PR TITLE
Fix SNMP infinite recursion & type detection

### PR DESCRIPTION
**Pull request by gpotter2**
- fixes https://github.com/secdev/scapy/issues/1639
- fixes https://github.com/secdev/scapy/issues/1901

### CODE REVIEW

File: scapy/asn1/ber.py [hunk 1]
```diff
@@ -294,7 +298,7 @@ def safedec(cls, s, context=None):
 
     @classmethod
     def enc(cls, s):
-        if isinstance(s, six.string_types):
+        if isinstance(s, (str, bytes)):
```
Comment (p-l-) [inline scapy/asn1/ber.py]: Why is this needed?

Comment (gpotter2) [reply scapy/asn1/ber.py]: That’s for https://github.com/secdev/scapy/issues/1901

It’s a bug from when it was ported to Python 3: it only expects str but should also expect bytes

Comment (p-l-) [reply scapy/asn1/ber.py]: Then we're probably better with this, to accept `unicode` values in Python 2 (as they are the equivalent of `str` in Python 3):
```suggestion
        if isinstance(s, six.string_types + (bytes,)):
```

**Comment by gpotter2** [comment]
The Travis CI failure is related to ISOTP. I’m trying to keep the build as-it so that it can be debugged
